### PR TITLE
Adds junk barrier -> junk wall

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary

```SUMMARY: Content "Adds shanty-town walls comprised of bolted-together junk"``` 

#### Purpose of change

The majority of makeshift construction in game currently uses wood, be it logs or two by fours. Although scrap wood is abundant in the cataclysm, so is scrap metal. Current metal walls (even though they're termed as "scrap metal walls" in the source file) are very sturdy, welded together walls made from durable steel plates, with very high HP. Where are the junky walls made from rusty corrugated sheet metal and bits of old cars held together by zip ties and loose bolts?

#### Describe the solution

The recipe for these walls allows a wide range of possible ingredients, to help them be true 'scrap walls' made from what is on hand. The recipe doesn't require a welder, but to help make up for this, does require a realistic wide range of other tools - in particular a wrench and a drill, to put holes where needed and shove bolts (represented by spikes), nails (representing metal screws which don't exist in game), or wires through.  The HP of these walls is enough to make them decent barriers against zombies, but not nearly as high as the higher tier walls. It is also not quite as tough as log palisades, although it fills a similar role. 

The first stage construction, junk metal barrier, can be reinforced with significantly more resources (still welder-free) to build a junk metal wall. This wall is comparable in durability to brick, and less durable than log or stone. A decent wall, but nothing game-breaking I think. The barrier, like the palisade wall, does not support a roof. The upgrade does.

#### Describe alternatives you've considered

I considered requiring a few welding charges for the reinforcement phase (4 acetylene, 20 regular, 30 makeshift/integrated) but decided that it defeated the purpose of the whole affair.

I'd consider an alternate reinforcement phase that used a welder and required fewer materials to make though. In fact I may include that before the PR is done. (Edit: talked myself into it and added)

#### Additional context

I intend to add a few more parts to this before the day is done - junk metal roofs and gates mainly. Probably not in this PR.

### CODE REVIEW

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/developing-a-fortified-evac-shelter-structure/17837/15

File: data/json/terrain.json [hunk 1]
```diff
@@ -7923,5 +7923,69 @@
       "ter_set": "t_pit_shallow",
       "items": [ { "item": "rock", "count": 8 }, { "item": "stick", "count": [ 8, 12 ] } ]
     }
+  },
+  {
+    "type": "terrain",
+    "id": "t_junk_palisade",
+    "name": "junk metal barrier",
+	"looks_like": "t_scrap_wall_halfway",
```
Comment (kevingranade) [inline data/json/terrain.json]:
```suggestion
    "looks_like": "t_scrap_wall_halfway",
```

File: data/json/terrain.json [hunk 2]
```diff
@@ -7923,5 +7923,69 @@
...
+      "sound_fail": "clang!",
+      "ter_set": "t_pit_shallow",
+      "items": [ { "item": "pipe", "count": [ 1, 2 ] }, { "item": "scrap", "count": [ 1, 40 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_junk_wall",
+    "name": "junk metal wall",
+	"looks_like": "t_scrap_wall",
```
Comment (kevingranade) [inline data/json/terrain.json]:
```suggestion
    "looks_like": "t_scrap_wall",
```

File: data/json/terrain.json [hunk 3]
```diff
@@ -7923,5 +7923,69 @@
...
+        { "item": "sheet_metal", "count": [ 1, 2 ] },
+        { "item": "steel_chunk", "count": [ 1, 30 ] }
+      ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_junk_floor",
+    "name": "junk metal floor",
+	"looks_like": "t_scrap_floor",
```
Comment (kevingranade) [inline data/json/terrain.json]:
```suggestion
    "looks_like": "t_scrap_floor",
```

File: data/json/construction.json [hunk 4]
```diff
@@ -2409,5 +2418,101 @@
...
+    "post_terrain": "t_junk_wall"
+  },
+  {
+    "type": "construction",
+    "description": "Reinforce Junk Metal Wall using spot-welds",
+    "category": "CONSTRUCT",
+    "required_skills": [ [ "fabrication", 4 ] ],
+    "time": 90,
+    "qualities": [ [ { "id": "SAW_M", "level": 1 } ], [ { "id": "GLARE", "level": 2 } ] ],
+    "tools": [ [ [ "oxy_torch", 4 ], [ "welder", 20 ], [ "welder_crude", 30 ], [ "toolset", 30 ] ] ],
```
Comment (Night-Pryanik) [inline data/json/construction.json]: You can replace this with `welding_standard` requirement.

File: data/json/terrain.json [hunk 5]
```diff
@@ -7923,5 +7923,69 @@
       "ter_set": "t_pit_shallow",
       "items": [ { "item": "rock", "count": 8 }, { "item": "stick", "count": [ 8, 12 ] } ]
     }
+  },
+  {
+    "type": "terrain",
+    "id": "t_junk_palisade",
+    "name": "junk metal barrier",
+    "looks_like": "t_scrap_wall_halfway",
+    "description": "A simple wall of rusty scrap metal bolted and wire-tied to a makeshift frame. Very fashionable in post-apocalyptic shantytowns. This one isn't quite strong enough to support a roof, but could be reinforced.",
```
Comment (Night-Pryanik) [inline data/json/terrain.json]:
```suggestion
    "description": "A simple wall of rusty scrap metal bolted and wire-tied to a makeshift frame.  Very fashionable in post-apocalyptic shantytowns.  This one isn't quite strong enough to support a roof, but could be reinforced.",
```

File: data/json/terrain.json [hunk 6]
```diff
@@ -7923,5 +7923,69 @@
...
+      "ter_set": "t_pit_shallow",
+      "items": [ { "item": "pipe", "count": [ 1, 2 ] }, { "item": "scrap", "count": [ 1, 40 ] } ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_junk_wall",
+    "name": "junk metal wall",
+    "looks_like": "t_scrap_wall",
+    "description": "A wall of rusty scrap metal bolted and wire-tied to a sturdy frame. Very fashionable in post-apocalyptic shantytowns. Can support a roof.",
```
Comment (Night-Pryanik) [inline data/json/terrain.json]:
```suggestion
    "description": "A wall of rusty scrap metal bolted and wire-tied to a sturdy frame.  Very fashionable in post-apocalyptic shantytowns.  Can support a roof.",
```

File: data/json/terrain.json [hunk 7]
```diff
@@ -7923,5 +7923,69 @@
...
+        { "item": "steel_chunk", "count": [ 1, 30 ] }
+      ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_junk_floor",
+    "name": "junk metal floor",
+    "looks_like": "t_scrap_floor",
+    "description": "A simple roof and floor of rusty scrap metal bolted and wire-tied to a makeshift frame. Very fashionable in post-apocalyptic shantytowns. Hope you like the sound of rain on corrugated metal.",
```
Comment (Night-Pryanik) [inline data/json/terrain.json]:
```suggestion
    "description": "A simple roof and floor of rusty scrap metal bolted and wire-tied to a makeshift frame.  Very fashionable in post-apocalyptic shantytowns.  Hope you like the sound of rain on corrugated metal.",
```

File: data/json/terrain.json (hunk 6 above)
Comment (ZhilkinSerg) [reply data/json/terrain.json]: Also two whitespaces before "Very".

**Pull request merged**